src/sonar_widget/SonarPlot.cc

```diff
@@ -276,20 +274,16 @@ void SonarPlot::generateMultibeamTransferTable(const base::samples::Sonar& sonar
             point.ry() /= scaleY * BINS_REF_SIZE / sonar.bin_count;
 
             double radius = sqrt(point.x() * point.x() + point.y() * point.y());
-            double angle = atan2(point.x(), -point.y());
+            base::Angle angle = base::Angle::fromRad( atan2(point.x(), -point.y()) );
 
             // pixels out of sonar image
-            if (angle < sonar.bearings[0].rad || angle > sonar.bearings[sonar.beam_count - 1].rad || radius > sonar.bin_count || !radius || j > origin.y())
+            if (angle < sonar.bearings[0] || angle > sonar.bearings[sonar.beam_count - 1] || radius > sonar.bin_count || !radius || j > origin.y())
                 transfer.push_back(-1);
 
             // pixels in the sonar image
             else {
-                for (unsigned int k = 0; k < sonar.beam_count - 1; k++) {
-                    if (angle >= sonar.bearings[k].rad && angle < sonar.bearings[k + 1].rad) {
-                        transfer.push_back(k * sonar.bin_count + radius);
-                        break;
-                    }
-                }
+                int idBeam = binarySearch(sonar.bearings, 0, sonar.bearings.size() - 1, angle);
```

Comment:
If you can do a binary search, it means you assume that the sonar bearings are sorted in increasing order (a fair assumption !). Which also means that the idBeam of a pixel on the line is either the same than the pixel before it, or the next one.

A.k.a.: you don't need to do search *at all*, and definitely have no need for binarySearch.